lib/eclipse/include/opm/parser/eclipse/EclipseState/Aquifer.hpp

```diff
@@ -0,0 +1,170 @@
+            m_influenceFnTableNum( 0 )
+        {
+	  Deck init_deck = Deck();
+	  deck_ = &init_deck;
+	}
+
+        explicit Aquifer(const Deck& deck) :
+            Aquifer()
+	    {
+	      deck_ = deck;
+	      if (deck.hasKeyword("AQUCT")) {
+		  const auto& record = deck.getKeyword( "AQUCT" , 0 ).getRecord( 0 );
+		  m_aquId  = record.getItem("AQUIFER_ID").get<int>(0);
+		  m_datDepth  = record.getItem("DAT_DEPTH").get<double>(0);
+		  m_aqPerm  = record.getItem("PERM_AQ").get<double>(0);
+		  m_aqPoro  = record.getItem("PORO_AQ").get<double>(0);
+		  m_aqCt  = record.getItem("C_T").get<double>(0);
+		  m_aqRad  = record.getItem("RAD").get<double>(0);
+		  m_aqThickness  = record.getItem("THICKNESS_AQ").get<double>(0);
+		  m_aqInfluenceAngle  = record.getItem("INFLUENCE_ANGLE").get<double>(0);
+		  m_waterPressTableNum  = record.getItem("TABLE_NUM_WATER_PRESS").get<int>(0);
+		  m_influenceFnTableNum  = record.getItem("TABLE_NUM_INFLUENCE_FN").get<int>(0);
+	      }
+	      
+	      int number_tables = m_aqudims.NIFTBL;
+	      
+	      
+            
+
+        
+        }
+
+        size_t getAquId() const
+        {
+            return m_aquId;
+        }
+
+        size_t getAqDatumDepth() const
+        {
+            return m_datDepth;
+        }
+
+        size_t getAqPermeability() const
+        {
+            return m_aqPerm;
+        }
+
+        size_t getAqPorosity() const
+        {
+            return m_aqPoro;
+        }
+
+        size_t getAqTotalCompressibility() const
+        {
+            return m_aqCt;
+        }
+
+        size_t getAqRadius() const
+        {
+            return m_aqRad;
+        }
+        
+        size_t getAqThickness() const
+        {
+            return m_aqThickness;
+        }
+
+        size_t getAqInfluenceAngle() const
+        {
+            return m_aqInfluenceAngle;
+        }
+
+        size_t getAqWaterPressureTableNumber() const
+        {
+            return m_waterPressTableNum;
+        }        
+
+        size_t getAqInfluenceFunctionTableNumber() const
+        {
+            return m_influenceFnTableNum;
+        }        
+      
+        std::vector<double> getTimeColumn() const{
+	if (deck_.hasKeyword("AQUTAB")) {
+            if (m_influenceFnTableNum == 1) {
+                  return default_time_;
+              }
+            else {
+                  return Aqutab_tables_.at(m_influenceFnTableNum-1).getTimeColumn();
+              }
+             
+	}
+	  
+	  
+	  
+        std::vector<double> getPressureColumn() const{
+        if (m_influenceFnTableNum == 1) {
+              return default_pressure_;
+          }
+        else {
+              return Aqutab_tables_.at(m_influenceFnTableNum-1).getPressureColumn();
+          }
+        }
+
+    private:
+        size_t m_aquId , m_datDepth , m_aqPerm , m_aqPoro , m_aqCt , m_aqRad , m_aqThickness , m_aqInfluenceAngle, m_waterPressTableNum, m_influenceFnTableNum;
+	Deck& deck_;
```

Comment:
No - you must find another solution - internalizing a private `Deck&` will not be merged.